pulsar-functions/proto/src/main/proto/Function.proto

```diff
@@ -94,6 +94,7 @@ message ConsumerSpec {
         int32 value = 1;
     }
     ReceiverQueueSize receiverQueueSize = 4;
+    bool readCompacted = 5;
```

Comment:
I think a more generic approach is to add a `Map<String, String>` to the consumer spec. So we can use https://github.com/apache/pulsar/blob/870a637b4906862a611e418341dd926e21458f08/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBuilderImpl.java#L86 to load the consumer configuration when building the consumers.

Reply:
We already have a ConsumerConfig class:

https://github.com/apache/pulsar/blob/master/pulsar-common/src/main/java/org/apache/pulsar/common/functions/ConsumerConfig.java

Though I am for refactoring this to be generic Map so we don't have to make code changes for new consumer configs that are added.

Reply:
> We already have a ConsumerConfig class:
> 
> https://github.com/apache/pulsar/blob/master/pulsar-common/src/main/java/org/apache/pulsar/common/functions/ConsumerConfig.java
> 
> Though I am for refactoring this to be generic Map so we don't have to make code changes for new consumer configs that are added.

Thanks very much. But schemaProperties are properties belonging to the schema. Put the Consumer parameters with them, I don’t think it’s a good way. I think it is necessary to add a new Map to save the general parameters of Consumer.And this make it easier to convert Map to object